source/extensions/common/wasm/context.cc

```diff
@@ -696,16 +696,18 @@ WasmResult Context::getHeaderMapValue(WasmHeaderMapType type, absl::string_view
     return WasmResult::BadArgument;
   }
   const Http::LowerCaseString lower_key{std::string(key)};
-  auto entry = map->get(lower_key);
-  if (!entry) {
+  const auto entry = map->get(lower_key);
+  if (entry.empty()) {
     if (wasm()->abiVersion() == proxy_wasm::AbiVersion::ProxyWasm_0_1_0) {
       *value = "";
       return WasmResult::Ok;
     } else {
       return WasmResult::NotFound;
     }
   }
-  *value = entry->value().getStringView();
+  // TODO(kyessenov, PiotrSikora): This needs to either return a concatenated list of values, or
+  // the ABI needs to be changed to return multiple values. This is a potential security issue.
```

Comment:
@kyessenov @PiotrSikora this needs to be fixed ASAP

Reply:
(Talked to @kyessenov offline about this and it will be fixed soon)

Reply:
FWIW, the ABI in proxy-wasm/spec#1 already returns multiple values.

Reply:
Perfect. Let's do that then.